File: MaWPhotos/src/main/java/us/mikeandwan/photos/di/HttpModule.java
package us.mikeandwan.photos.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import us.mikeandwan.photos.services.AuthAuthenticator;
import us.mikeandwan.photos.services.AuthInterceptor;


@Module
public class HttpModule {
    @Provides
    @Singleton
    OkHttpClient provideOkHttpClient(AuthAuthenticator authenticator, AuthInterceptor authInterceptor) {
        return new OkHttpClient
            .Builder()
            .authenticator(authenticator)
            .addInterceptor(authInterceptor)
            .build();
    }
}

File: MaWPhotos/src/main/java/us/mikeandwan/photos/di/HttpModule.kt
package us.mikeandwan.photos.di

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import us.mikeandwan.photos.services.AuthAuthenticator
import us.mikeandwan.photos.services.AuthInterceptor
import okhttp3.OkHttpClient
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class HttpModule {
    @Provides
    @Singleton
    fun provideOkHttpClient(
        authenticator: AuthAuthenticator?,
        authInterceptor: AuthInterceptor?
    ): OkHttpClient {
        return OkHttpClient.Builder()
            .authenticator(authenticator)
            .addInterceptor(authInterceptor)
            .build()
    }
}